packages/dataviews/src/pagination.js

```diff
@@ -27,36 +27,33 @@ const Pagination = memo( function Pagination( {
 				justify="end"
 				className="dataviews-pagination"
 			>
-				<HStack justify="flex-start" expanded={ false } spacing={ 2 }>
-					{ createInterpolateElement(
-						sprintf(
+				<HStack justify="flex-start" expanded={ false } spacing={ 1 }>
+					<SelectControl
+						aria-label={ __( 'Current page' ) }
+						value={ view.page }
+						className="dataviews-pagination__page-selector"
+						options={ Array.from( Array( totalPages ) ).map(
+							( _, i ) => {
+								const page = i + 1;
+								return { value: page, label: page };
+							}
+						) }
+						onChange={ ( newValue ) => {
+							onChangeView( {
+								...view,
+								page: +newValue,
+							} );
+						} }
+						size={ 'compact' }
```

Comment:
No need to interpolate a raw string

```suggestion
						size="compact"
```